### ISSUE TITLE
Miner cannot use custom node schematic

### ISSUE THREAD

**Issue by leagris** [ldtteam/minecolonies#7539]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 0.14.350-ALPHA
- Structurize Version: 0.13.210-ALPHA
- Related Mods and their Versions: -

## Expected behavior
Miner use custom mine nodes schematics as long as these use proper dimensions and orientations

## Actual behavior
Miner stuck at level after building the first 3 straight tunnels from the main shaft.
![2021-08-27_09 54 19](https://user-images.githubusercontent.com/1111474/131093150-5ba70704-0cb8-49e7-9347-22a6603e59b8.png)

## Steps to reproduce the issue

1. Download and extract these miner nodes schematics https://www.noiraude.net/bug/miner.7z
in: `minecraft/structurize/schematics/stone/` or any other style you want to overrides the mine nodes.
2. Build mine in style using these new nodes schematics.
3. Let miner build a mineshaft and start digging nodes.
4. See miner stuck after finishing the third node.

### Logs

- latest.log: recurrent spam of:  
  ```none
   [27Aug2021 09:36:34.581] [Server thread/WARN] [structurize/]: Could not find iterator for value: using default instead!
   ```
- crashlog: 

### Notes

- Custom mine nodes schematics strictly respect size, orientation and are exempt from entities.
- Created exclusively with Minecolonies and Structurize mods.
- Only use these blocks:
  - cobblestone
  - cobblestone wall
  - cobblestone slab
  - cobblestone stair
  - torch
  - solid block placeholder

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by uecasm** [comment]
Have you tried removing the roof?  Maybe the tunnel is too low for them.

**Comment by leagris** [comment]
> Have you tried removing the roof? Maybe the tunnel is too low for them.

Custom mine **nodes schematics strictly respect size. **

They are exactly same as the reference schematics, except:
- use a center pillar of two cobblestone walls and a cobblestone with torches
- a partial arch of two cobblestone stairs and a cobblestone slab at exit side

rather than arches made of two pillars of two cobblestone walls headed by a 3 cobblestone lintel holding a torch.

Miner perfectly builds the first three nodes, then get stuck.

If I break or alter an already built nodes, miner immediately repairs it, but still fail stuck at the main shaft when finished repairing the altered node.

**Comment by leagris** [comment]
Fired and re hired miner and miner immediately returned stuck at the main shaft, not even taking tools or blocks with him.

What this tells is that the miner's AI state machine is stuck at the planning stage.

@Raycoms 
Literally the miner AI fail to choose the next node to place.

**Comment by uecasm** [comment]
And my point is that in your screenshot they look exactly like they're trying to enter the shaft but can't fit.  And the cobble roof looks like it's too low.  So like I said, try removing it and see what happens.  Just break it in-world, you don't need to change the schematic.

**Comment by leagris** [comment]
> And my point is that in your screenshot they look exactly like they're trying to enter the shaft but can't fit. And the cobble roof looks like it's too low. So like I said, try removing it and see what happens. Just break it in-world, you don't need to change the schematic.

They do enter the shaft to repair it if I break or add blocks in it. At least they don't seem to have an issue to enter for repairing. The Guard does enter occasionally.

**Comment by leagris** [comment]
Finally identified the cause of it and it:

Miner check for the presence of cobblestone in the way of the next node placement, this to avoid digging a node through the main shaft's cobblestone ladder backbone. The main shaft cobblestone backbone is detected because of its cobblestone blocks.

My custom miner node schematic had a pillar in the middle of the node, made of two cobblestone walls and a cobblestone block. The miner detected this pillar as a deny to create a node. Once the 3 `minerx2top` nodes starting from the main shaft were built, the miner has a deny to create a node in all directions, so was stuck there, unable to plan new nodes.

This issue has the very same cause as: [Miner blocked by cobblestone #5483](https://github.com/ldtteam/minecolonies/issues/5483)
